packages/react-ui/src/TableOfContent/useTOCReducer.js

```diff
@@ -0,0 +1,32 @@
+import { useReducer, useCallback } from "react";
+
+export const rootReducer = (state, action) => {
+  switch (action.type) {
+    case "setTitles":
+      return { titles: [...action.payload.map((title) => ({ ...title }))] };
+    case "setActive":
+      return {
+        titles: [
+          ...state.titles.map((title) => ({
```

Comment:
destructuring is not needed here. just use 
```js
titles: state.titles.map((title) => {
  title.active = title.id === action.payload 
  return title;
})
```

Reply:
I don't really like to modifiy the initial array too even if a new one is created. But the two array references the same objects which is bug prone.
Is this ok ?
```
        titles: state.titles.map((title) => ({
          ...title,
          active: title.id === action.payload,
        })),
```
?

Reply:
yup but destructuring title only work for scalar value, it is not deep clone thats why I think it not usefull

Reply:
for now `title` only holds scalars so I guess that's ok. Maybe cast an eye on deep clones utilities later on